build-scripts/ci_parser/ci_parser.py

```diff
@@ -68,50 +85,50 @@ def main():
     """
 
     # Skip archives that we know exist but don't have results we want.
-    gz_exclusions = ['split', 'result_details']
+    xml_exclusions = ['split', 'result_details']
 
     # Inclusions win over exclusions right now but are empty by default.
 
     # TODO: Make this a command-line regex? Used for debugging; could use to parse just a certain subset of suites.
-    gz_inclusions = None  # type: Optional[List[str]]
-    # gz_inclusions = ['python']
+    xml_inclusions = None  # type: Optional[List[str]]
+    # xml_inclusions = ['python']
 
     # TODO: Make this a command-line debug flag? Used for debugging
     debug_file = None  # type: Optional[str]
-    # debug_file = 'jvm17-utests_archive.tar.gz'
+    # debug_file = 'jvm17-utests_archive.tar.xml'
     if debug_file is not None:
-        gz_files = [str(file) for file in Path(input_dir).rglob('*.gz') if debug_file in str(file)]
-    elif gz_inclusions is not None:
-        gz_files = [str(file) for file in Path(input_dir).rglob('*.gz') if any(x in str(file) for x in gz_inclusions)]
+        xml_files = [str(file) for file in Path(input_dir).rglob('*.xml') if debug_file in str(file)]
+    elif xml_inclusions is not None:
+        xml_files = [str(file) for file in Path(input_dir).rglob('*.xml') if any(x in str(file) for x in xml_inclusions)]
     else:
-        gz_files = [str(file) for file in Path(input_dir).rglob('*.gz') if not any(x in str(file) for x in gz_exclusions)]
-    check_file_condition(lambda: len(gz_files) != 0, f'Found 0 .gz files in path: {input_dir}. Cannot proceed with .xml extraction.')
-    logger.debug(f'Extracting .xml from {len(gz_files)} gzip files from path: {input_dir}')
+        xml_files = [str(file) for file in Path(input_dir).rglob('*.xml') if not any(x in str(file) for x in xml_exclusions)]
+    check_file_condition(lambda: len(xml_files) != 0, f'Found 0 .xml files in path: {input_dir}. Cannot proceed with .xml extraction.')
+    logger.debug(f'Extracting .xml from {len(xml_files)} xml files from path: {input_dir}')
 
     archive_count = 0
     test_file_count = 0
     test_count = 0
 
     test_suites = dict()  # type: Dict[str, JUnitTestSuite]
 
-    logger.info('List of gzip files to be processed:')
-    for file in gz_files:
+    logger.info('List of xml files to be processed:')
+    for file in xml_files:
         logger.info(f' -- {file}')
 
-    # Since we have a 1:1 ratio on .gz files to suites, we can parallelize w/out any kind of synchronization. We also
+    # Since we have a 1:1 ratio on .xml files to suites, we can parallelize w/out any kind of synchronization. We also
     # check to ensure this contract is upheld in the processing method.
     with ThreadPoolExecutor() as executor:
-        futures = [executor.submit(process_gzip_file, gz_file, input_dir, test_suites) for gz_file in gz_files]
+        futures = [executor.submit(process_xml_file, xml_file, input_dir, test_suites) for xml_file in xml_files]
```

Comment:
So in this case here, do we have one monolithic xml file, or are you about to spawn 1 thread per .xml file? That'd be... unnecessarily wasteful imo.

If we have _directories_ we're operating against, we could use those in lieu of the .gz files. That way we could at least still parallelize safely across directories. Otherwise, I'd recommend removing the parallelism entirely in this configuration and just serially go across all .xml files.

As it stands, this logging and flow is no longer accurate.

Reply:
can be many thousands of files.  a limited pool of threads would be useful, but how to do that in python isn't top of mind…